### PR TITLE
exporter: add logging, error handling and some cleanups

### PR DESCRIPTION
**Pull request by avanthakkar**
The PR addresses some minor fixes, logging for update and dump daemon socket files. Also it includes some error handling for `async_read` and `write_response` of `http_server`.

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "pacific"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [ ] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [ ] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

**Comment by avanthakkar** [comment]
jenkins test api

**Comment by avanthakkar** [comment]
jenkins test docs

File: src/exporter/DaemonMetricCollector.cc [hunk 1]
```diff
@@ -96,10 +96,12 @@ void DaemonMetricCollector::dump_asok_metrics() {
     }
     std::string perf_dump_response = asok_request(sock_client, "perf dump");
     if (perf_dump_response.size() == 0) {
+      std::cout << "Perf counters dump retrieval for daemon " << daemon_name << " failed." << std::endl;
```
Comment (epuertat) [inline src/exporter/DaemonMetricCollector.cc:99]: Doesn't the call above provide any reason why this failed? Error code?

File: src/exporter/DaemonMetricCollector.cc [hunk 2]
```diff
@@ -137,6 +139,7 @@ void DaemonMetricCollector::dump_asok_metrics() {
       }
     }
   }
+  std::cout << "Perf counters retrieved for " << clients.size() << " daemons." << std::endl;
```
Comment (epuertat) [inline src/exporter/DaemonMetricCollector.cc:142]: What about summarizing the successful ones vs the total configured?

**Review by epuertat — APPROVED** [review]
Nice! Just left some comments over there. Can you please paste the output of this log with some intended failures? Thanks!

LGMT anyway! Thanks @avanthakkar !

File: src/exporter/DaemonMetricCollector.cc (hunk 2 above)
Comment (pereman2) [reply src/exporter/DaemonMetricCollector.cc:144]:
```suggestion
  dout(level) << "Perf counters retrieved for " << clients.size() << " daemons." << dendl;
```

**Comment by avanthakkar** [comment]
jenkins retest this please

**Review by pereman2 — APPROVED** [review]

File: src/exporter/DaemonMetricCollector.cc [hunk 3]
```diff
@@ -221,11 +226,13 @@ void DaemonMetricCollector::get_process_metrics(std::vector<std::pair<std::strin
...
 
 std::string DaemonMetricCollector::asok_request(AdminSocketClient &asok,
-                                                std::string command) {
+                                                std::string command, std::string daemon_name) {
   std::string request("{\"prefix\": \"" + command + "\"}");
   std::string response;
   std::string err = asok.do_request(request, &response);
   if (err.length() > 0 || response.substr(0, 5) == "ERROR") {
+    std::cout << command << "command failed for daemon " << daemon_name << std::endl;
+    dout(1) << err << dendl;
```
[the remark this answers is not in the document]
Comment (avanthakkar) [reply src/exporter/DaemonMetricCollector.cc:235]: I just followed the critical < error < debug < info hierarchy for log levels (assuming levels to be 0, 1, 5, 10 respectively)

**Comment by avanthakkar** [comment]
jenkins test make check

**Pull request merged**